os/binfmt/binfmt_execmodule.c

```diff
@@ -272,20 +269,23 @@ int exec_module(FAR struct binary_s *binp)
 
 	pid = newtcb->cmn.pid;
 
-	/* Parent task rtcb was initialized with uheap and mpu_regs values.
-	 * These values are used for stack allocation and task_init and will
-	 * be passed on to all the child tasks of the user app. Since the user
-	 * app is now created, we will reset these values here
-	 */
-	rtcb->uheap = 0;
-#ifdef CONFIG_ARM_MPU
-	memset(rtcb->mpu_regs, 0, sizeof(rtcb->mpu_regs));
-#endif
-
 	/* Store the address of the applications userspace object in the newtcb  */
 	/* The app's userspace object will be found at an offset of 4 bytes from the start of the binary */
 	newtcb->cmn.uspace = binp->sections[BIN_TEXT] + 4;
 	newtcb->cmn.uheap = (uint32_t)binp->uheap;
+	/* Copy the MPU register values from parent to child task */
+#ifdef CONFIG_ARM_MPU
+	int i = 0;
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	for (; i < MPU_REG_NUMBER * MPU_NUM_REGIONS; i += MPU_REG_NUMBER)
+#endif
+	{
+		newtcb->mpu_regs[i + MPU_REG_RNR] = binp->mpu_regs[i + MPU_REG_RNR];
```

Comment:
`newtcb` is `struct task_tcb_s` type, so you should use `newtcb.cmn.mpu_regs` instead of `newtcb.mpu_regs`.

Reply:
Fixed